File: articles/alan-turing-computation-and-the-machine-model-of-reasoning/typescript/unary_increment.ts
function unaryIncrement(input: string): string {
  const tape = input.split("");
  let i = 0;
  while (i < tape.length && tape[i] === "1") {
    i += 1;
  }
  if (i === tape.length) {
    tape.push("_");
  }
  tape[i] = "1";
  if (i + 1 === tape.length) {
    tape.push("_");
  }
  return tape.join("");
}

console.log(`incremented_tape=${unaryIncrement("111_")}`);

File: articles/alan-turing-computation-and-the-machine-model-of-reasoning/go/unary_increment.go
package main

import "fmt"

func unaryIncrement(input string) string {
	tape := []rune(input)
	i := 0
	for i < len(tape) && tape[i] == '1' {
		i++
	}
	if i == len(tape) {
		tape = append(tape, '_')
	}
	tape[i] = '1'
	if i+1 == len(tape) {
		tape = append(tape, '_')
	}
	return string(tape)
}

func main() {
	fmt.Println("incremented_tape=" + unaryIncrement("111_"))
}
